jujubundlelib/references.py

```diff
@@ -181,13 +171,12 @@ def copy(self, **kwargs):
     def jujucharms_id(self):
```

Comment:
Why not accepting a channel here so that it's easier to generate the full URL?
like ```def jujucharms_id(self, channel=None):```
And then, if channel is not None, append the ?channel query.
This way we have a single place where this logic is defined.
Also, below, for the jujucharms_url method:
```python
def jujucharms_url(self, channel=None):
          """Return the URL where this entity lives in jujucharms.com."""
         return JUJUCHARMS_URL + self.jujucharms_id(channel=channel)
```